backend/src/src-cvesearch/src/main/java/com/bosch/osmi/sw360/cvesearch/helper/VulnerabilityUtils.java

```diff
@@ -0,0 +1,89 @@
+
+import com.siemens.sw360.datahandler.common.CommonUtils;
+import com.siemens.sw360.datahandler.thrift.RequestStatus;
+import com.siemens.sw360.datahandler.thrift.cvesearch.UpdateType;
+import com.siemens.sw360.datahandler.thrift.cvesearch.VulnerabilityUpdateStatus;
+import com.siemens.sw360.datahandler.thrift.vulnerabilities.Vulnerability;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.siemens.sw360.datahandler.common.CommonUtils.nullToEmptyList;
+
+public class VulnerabilityUtils {
+
+    public static VulnerabilityUpdateStatus getEmptyVulnerabilityUpdateStatus(RequestStatus requestStatus){
+        VulnerabilityUpdateStatus status = new VulnerabilityUpdateStatus();
+        status.statusToVulnerabilityIds = new HashMap<>();
+        for(UpdateType updateType: UpdateType.values()){
+            status.statusToVulnerabilityIds.put(updateType, new ArrayList<>());
+        }
+        status.requestStatus = requestStatus;
+        return status;
+    }
+
+    public static VulnerabilityUpdateStatus getEmptyVulnerabilityUpdateStatus(){
+        Map<UpdateType, List<String>> updateTypeToExternalID = new HashMap<>();
+        for (UpdateType updateType: UpdateType.values()){
+            updateTypeToExternalID.put(updateType, new ArrayList<>());
+        }
+        return new VulnerabilityUpdateStatus()
+                .setRequestStatus(RequestStatus.SUCCESS)
+                .setStatusToVulnerabilityIds(updateTypeToExternalID);
+    }
+
+    public static VulnerabilityUpdateStatus reduceVulnerabilityUpdateStatus(VulnerabilityUpdateStatus ... statuses){
+        VulnerabilityUpdateStatus resultStatus = getEmptyVulnerabilityUpdateStatus();
+
+        RequestStatus requestStatus = Arrays.stream(statuses)
+                .map(VulnerabilityUpdateStatus::getRequestStatus)
+                .reduce(RequestStatus.SUCCESS, CommonUtils::reduceRequestStatus);
+        resultStatus.setRequestStatus(requestStatus);
+
+        for(UpdateType updateType : UpdateType.values()) {
+            for (VulnerabilityUpdateStatus status : statuses) {
+                if(status.isSetStatusToVulnerabilityIds()) {
+                    resultStatus.statusToVulnerabilityIds.get(updateType)
+                            .addAll(nullToEmptyList(status.statusToVulnerabilityIds.get(updateType)));
+                }
+            }
+        }
+        return resultStatus;
+    }
+
+    public static VulnerabilityUpdateStatus getUpdateStatusFromUpdateMap(Map<UpdateType, List<Vulnerability>> statusToVulnerabilities){
+        VulnerabilityUpdateStatus status = getEmptyVulnerabilityUpdateStatus();
+        status.requestStatus = nullToEmptyList(statusToVulnerabilities.get(UpdateType.FAILED)).size() > 0
+                ? RequestStatus.FAILURE
+                : RequestStatus.SUCCESS;
+
+        for(UpdateType updateType: UpdateType.values()){
+            List<String> idList = nullToEmptyList(statusToVulnerabilities.get(updateType))
```

Comment:
If I read it correctly, `statusToVulnerabilities` is always pre-filled with empty lists, or at least supposed to be. Thus, `nullToEmptyList` shouldn't be necessary here.